jlm/hls/backend/rvsdg2rhls/rvsdg2rhls.cpp

```diff
@@ -451,9 +452,10 @@ dump_ref(llvm::RvsdgModule & rhls, std::string & path)
   instrument_ref(*reference);
   for (size_t i = 0; i < reference->Rvsdg().root()->narguments(); ++i)
   {
-    auto arg = reference->Rvsdg().root()->argument(i);
-    auto imp = dynamic_cast<const llvm::impport *>(&arg->port());
```

Comment:
This right here is the code that fails in the HLS tests in #578. The changes you make in this PR would also fix that issue (getting rid of ports sounds great). I also appreciate that the new code uses AssertedCast.